bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/ForecastSolarObject.java

```diff
@@ -46,6 +46,7 @@
  * The {@link ForecastSolarObject} holds complete data for forecast
  *
  * @author Bernd Weymann - Initial contribution
+ * @author Bernd Weymann - TimeSeries delivers only future values, otherwise past values are overwritten
```

Comment:
Can you describe the logic - why is it a problem that historic values are overwritten? Is it because the estimates provided by the API are updated to become more and more accurate, or possibly that historic values are becoming actual based on some sort of feedback? How many historic values are provided by the API?

Reply:
> Can you describe the logic - why is it a problem that historic values are overwritten?

It's described in the attached problem report #16885 
Problem describes updates back 2 days in the past. But in general a forecast values for e.g. 16:00 shouldn't be updated afterwards. I personally check also accuarcy values and they shoudln't change for past values.

> * First start: No historic values are persisted.

Correct. Before past values were in the timeseries which is not longer the case. Shouldn't be a problem: all data starts after Thing has been started. Other forecast bindings like openweather are doing the same.

> * Outage/downtime (cloud service, openHAB itself, network, etc.): Depending on duration, values could either be missing, or they could be inaccurate.

Correct. But this applies to all forecast data, right? If you take openweather precipitation and 1 day downtime the values  in openhab are one day old and not reflecting update in between. There's no corrextion afterwards.